### ISSUE TITLE
MA0075 and MA0185 are fighting

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#1310]
### Version of the Meziantou.Analyzer NuGet package

3.0.173

### Rule Identifier

MA0075

### Target Framework

.NET 10

### C# Language version

C# 14 (5.9.0-1.26379.115 (14fbf8d5271c98133561eb55185fdb05b286f578))

### .NET SDK version

10.0.400

### Description

If I'm not mistaken, all `ToString(...)` overloads are culture invariant.
In recent versions of *Meziantou.Analyzer*, **MA0075** no longer considers them invariant (while **MA0185** does).

### Reproduction Steps

Minimal code:

```c#
using System;
using System.Globalization;

var comparison = StringComparison.Ordinal;

_ = "ABC" + comparison.ToString("G"); // MA0075: Do not use implicit culture-sensitive ToString
_ = string.Create(CultureInfo.InvariantCulture, $"ABC '{comparison:G}"); // MA0185: Simplify string.Create when all parameters are culture invariant
```

**Comment by meziantou** [comment]
> If I'm not mistaken, all ToString(...) overloads are culture invariant.

Most of the time ;) `((MyEnum)-1).ToString()`. The `-` sign is culture-sensitive, like `-1.ToString()`.

That's being said, I think enum should be considered as culture-insensitive to reduce the number of false-positive. Also, `Enum.ToString(IFormatProvider)` doesn't use the format provider. Maybe a regression from the recent changes.

**Comment by drieseng** [comment]
@meziantou, thanks!
